presto-main/src/main/java/io/prestosql/sql/planner/SubqueryPlanner.java

```diff
@@ -297,6 +318,30 @@ private PlanBuilder appendExistSubqueryApplyNode(PlanBuilder subPlan, ExistsPred
                 correlationAllowed);
     }
 
+    private PlanBuilder appendArraySubqueryApplyNode(PlanBuilder subPlan, ArrayConstructor arraySubquery, boolean correlationAllowed)
```

Comment:
Not sure what I need to do here. Can I wrap the projection with an ARRAY_AGG here, or do I need to do the desugar later during PlanOptimizers?

Reply:
The way we deal with these (IN, EXISTS, etc) today is not great, but let's model it like those: we add an `Apply` node whose "assignment expression" is rewritten to be "x = ARRAY (y)", where y is the symbol name associated with the output of the subquery. So, what you have here is good.

We'll also need a rule similar to `TransformUncorrelatedInPredicateSubqueryToSemiJoin` that converts an Apply containing an Array(..) expression into a join + scalar subquery with with `array_agg`.